inference-engine/tests/unit/gna/pwl.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#include "runtime/pwl.h"
+#include "common_test_utils/data_utils.hpp"
+
+namespace {
+class PwlTest : public ::testing::Test {
+protected:
+    size_t range_search(double arg, const std::vector<pwl_t>& pwl) {
+        size_t left = 0;
+        size_t right = pwl.size() - 1;
+        size_t mid = (right - left) / 2;
+        while (left < right && (arg < pwl[mid].alpha || pwl[mid + 1].alpha < arg)) {
+            if (arg < pwl[mid].alpha) {
+                right = mid;
+            } else {
+                left = mid;
+            }
+
+            mid = left + (right - left) / 2;
+        }
+
+        return mid;
+    }
+
+    double pwl_function(double x, const std::vector<pwl_t>& pwl) {
+        size_t segment_index = 0;
+        if (x < pwl.front().alpha) {
+            segment_index = 0;
+        } else if (x > pwl.back().alpha) {
+            segment_index = pwl.size() - 2;
+        } else {
+            auto index = range_search(x, pwl);
+            segment_index = index == pwl.size() - 1 ? pwl.size() - 2 : index;
+        }
+
+        return pwl[segment_index].m * x + pwl[segment_index].b;
+    }
+
+    double function(double x, const DnnActivation& activation_type) {
+        switch (activation_type) {
+            case kActNegLog:
+                return 1.0 * log(x);
+            case kActNegHalfLog:
+                return 0.5 * log(x);
```

Comment:
shouldn't it be -1.0 * log(x) and -0.5 * log(x) ?

Reply:
done